src/adapter/properties.rs

```diff
@@ -377,6 +385,15 @@ pub(super) fn resolve_return_value_property<'a, V: AsVertex<Vertex<'a>> + 'a>(
                 .is_none()
                 .into()
         }),
+        "type" => resolve_property_with(contexts, |vertex| {
+            vertex
+                .as_return_value()
+                .expect("not a return value")
+                .type_
+                .map(rust_type_name::rust_type_name)
+                .unwrap_or("()".to_string())
+                .into()
```

Comment:
Let's try to avoid this `to_string()` call, because it'll be relatively common and we'll end up allocating twice (`.into()` will allocate again).

```suggestion
                .map(rust_type_name::rust_type_name)
                .map(Into::into)
                .unwrap_or("()".into())
```
Something like this should do the trick.